### PR TITLE
fix: disable ssl verification

### PR DESCRIPTION
**Pull request by vinayakkulkarni**
- fixes issue with RDS

### Changes

https://elixirforum.com/t/ecto-ssl-verification-with-google-sql-connection-requires-a-valid-client-certificate/4931

### CODE REVIEW

File: config/runtime.exs [hunk 1]
```diff
@@ -271,7 +271,8 @@ if is_nil(db_socket_dir) do
 else
   config :plausible, Plausible.Repo,
     socket_dir: db_socket_dir,
-    database: get_var_from_path_or_env(config_dir, "DATABASE_NAME", "plausible")
+    database: get_var_from_path_or_env(config_dir, "DATABASE_NAME", "plausible"),
+    ssl: false
```
Comment (vinayakkulkarni) [inline config/runtime.exs:275]: I think we have to https://elixirforum.com/t/aws-rds-castore-certificate-validation-for-aws-rds-dbs/55111

Comment (ruslandoga) [reply config/runtime.exs:275]: AFAIK TLS is used in Plausible Cloud so we won't be able to disable it for all setups.

Comment (ruslandoga) [reply config/runtime.exs:275]: I think it's already possible with `DATABASE_URL` like `ecto://postgres:postgres@host:5432/database?ssl=false`

**Comment by ruslandoga** [comment]
Your link redirects me to https://elixirforum.com/t/university-of-kent-courses-on-erlang-with-certification/4931 for some reason.

File: config/runtime.exs (hunk 1 above)
Comment (vinayakkulkarni) [reply config/runtime.exs:275]: Maybe we can pass in multiple arguments for building the DATABASE_URL ?

DB_HOST
DB_USER
DB_PORT
DB_NAME 

?

Comment (ruslandoga) [reply config/runtime.exs:275]: Why would it be better?

File: config/runtime.exs [hunk 2]
```diff
@@ -271,7 +271,8 @@ if is_nil(db_socket_dir) do
 else
   config :plausible, Plausible.Repo,
     socket_dir: db_socket_dir,
```
Comment (ruslandoga) [inline config/runtime.exs:273]: I noticed just now that you are configuring the repos that connect via unix sockets.

**Comment by vinayakkulkarni** [comment]
Trying to debug this:

```
2023-08-06 22:29:25 Loading plausible..
2023-08-06 22:29:25 Starting dependencies..
2023-08-06 22:29:26 Starting repos..
2023-08-06 22:29:26 Running migrations for Elixir.Plausible.Repo
2023-08-06 22:29:26 16:59:26.148 [error] Postgrex.Protocol (#PID<0.164.0>) failed to connect: ** (Postgrex.Error) FATAL 28000 (invalid_authorization_specification) no pg_hba.conf entry for host "103.198.165.143", user "postgres", database "analytics", no encryption
2023-08-06 22:29:26 16:59:26.148 [error] Postgrex.Protocol (#PID<0.163.0>) failed to connect: ** (Postgrex.Error) FATAL 28000 (invalid_authorization_specification) no pg_hba.conf entry for host "103.198.165.143", user "postgres", database "analytics", no encryption
2023-08-06 22:29:28 16:59:28.081 [error] Postgrex.Protocol (#PID<0.163.0>) failed to connect: ** (Postgrex.Error) FATAL 28000 (invalid_authorization_specification) no pg_hba.conf entry for host "103.198.165.143", user "postgres", database "analytics", no encryption
2023-08-06 22:29:29 16:59:29.030 [error] Postgrex.Protocol (#PID<0.164.0>) failed to connect: ** (Postgrex.Error) FATAL 28000 (invalid_authorization_specification) no pg_hba.conf entry for host "103.198.165.143", user "postgres", database "analytics", no encryption
2023-08-06 22:29:29 16:59:29.046 [error] Could not create schema migrations table. This error usually happens due to the following:
2023-08-06 22:29:29 
2023-08-06 22:29:29   * The database does not exist
2023-08-06 22:29:29   * The "schema_migrations" table, which Ecto uses for managing
2023-08-06 22:29:29     migrations, was defined by another library
2023-08-06 22:29:29   * There is a deadlock while migrating (such as using concurrent
2023-08-06 22:29:29     indexes with a migration_lock)
2023-08-06 22:29:29 
2023-08-06 22:29:29 To fix the first issue, run "mix ecto.create".
2023-08-06 22:29:29 
2023-08-06 22:29:29 To address the second, you can run "mix ecto.drop" followed by
2023-08-06 22:29:29 "mix ecto.create". Alternatively you may configure Ecto to use
2023-08-06 22:29:29 another table and/or repository for managing migrations:
2023-08-06 22:29:29 
2023-08-06 22:29:29     config :plausible, Plausible.Repo,
2023-08-06 22:29:29       migration_source: "some_other_table_for_schema_migrations",
2023-08-06 22:29:29       migration_repo: AnotherRepoForSchemaMigrations
2023-08-06 22:29:29 
2023-08-06 22:29:29 The full error report is shown below.
2023-08-06 22:29:29 
2023-08-06 22:29:39 Loading plausible..
2023-08-06 22:29:39 Starting dependencies..
2023-08-06 22:29:39 Starting repos..
2023-08-06 22:29:39 Running migrations for Elixir.Plausible.Repo
2023-08-06 22:29:39 16:59:39.674 [error] Postgrex.Protocol (#PID<0.164.0>) failed to connect: ** (Postgrex.Error) FATAL 28000 (invalid_authorization_specification) no pg_hba.conf entry for host "103.198.165.143", user "postgres", database "analytics", no encryption
2023-08-06 22:29:39 16:59:39.674 [error] Postgrex.Protocol (#PID<0.163.0>) failed to connect: ** (Postgrex.Error) FATAL 28000 (invalid_authorization_specification) no pg_hba.conf entry for host "103.198.165.143", user "postgres", database "analytics", no encryption
2023-08-06 22:29:41 16:59:41.584 [error] Postgrex.Protocol (#PID<0.163.0>) failed to connect: ** (Postgrex.Error) FATAL 28000 (invalid_authorization_specification) no pg_hba.conf entry for host "103.198.165.143", user "postgres", database "analytics", no encryption
2023-08-06 22:29:42 16:59:42.200 [error] Postgrex.Protocol (#PID<0.164.0>) failed to connect: ** (Postgrex.Error) FATAL 28000 (invalid_authorization_specification) no pg_hba.conf entry for host "103.198.165.143", user "postgres", database "analytics", no encryption
2023-08-06 22:29:42 16:59:42.644 [error] Could not create schema migrations table. This error usually happens due to the following:
2023-08-06 22:29:42 
2023-08-06 22:29:42   * The database does not exist
2023-08-06 22:29:42   * The "schema_migrations" table, which Ecto uses for managing
2023-08-06 22:29:42     migrations, was defined by another library
2023-08-06 22:29:42   * There is a deadlock while migrating (such as using concurrent
2023-08-06 22:29:29 ** (DBConnection.ConnectionError) connection not available and request was dropped from queue after 2975ms. This means requests are coming in and your connection pool cannot serve them fast enough. You can address this by:
2023-08-06 22:29:29 
2023-08-06 22:29:29   1. Ensuring your database is available and that you can connect to it
2023-08-06 22:29:29   2. Tracking down slow queries and making sure they are running fast enough
2023-08-06 22:29:29   3. Increasing the pool_size (although this increases resource consumption)
2023-08-06 22:29:29   4. Allowing requests to wait longer by increasing :queue_target and :queue_interval
2023-08-06 22:29:29 
2023-08-06 22:29:29 See DBConnection.start_link/2 for more information
2023-08-06 22:29:29 
2023-08-06 22:29:29     (ecto_sql 3.9.2) lib/ecto/adapters/sql.ex:913: Ecto.Adapters.SQL.raise_sql_call_error/1
2023-08-06 22:29:29     (elixir 1.14.3) lib/enum.ex:1658: Enum."-map/2-lists^map/1-0-"/2
2023-08-06 22:29:29     (ecto_sql 3.9.2) lib/ecto/adapters/sql.ex:1005: Ecto.Adapters.SQL.execute_ddl/4
2023-08-06 22:29:29     (ecto_sql 3.9.2) lib/ecto/migrator.ex:677: Ecto.Migrator.verbose_schema_migration/3
2023-08-06 22:29:29     (ecto_sql 3.9.2) lib/ecto/migrator.ex:491: Ecto.Migrator.lock_for_migrations/4
2023-08-06 22:29:29     (ecto_sql 3.9.2) lib/ecto/migrator.ex:403: Ecto.Migrator.run/4
2023-08-06 22:29:29     (ecto_sql 3.9.2) lib/ecto/migrator.ex:146: Ecto.Migrator.with_repo/3
2023-08-06 22:29:29     nofile:1: (file)
```

**Comment by guatedude2** [comment]
I'm still not able to move through this issue. I tried disabling SSL, I tried providing the AWS cert CA bundle and I still get this error:

With SSL disabled:
```
./createdb.sh 
Loading plausible..
Starting dependencies..
Starting repos..
create Plausible.Repo database if it doesn't exist
16:37:24.465 [error] Postgrex.Protocol (#PID<0.185.0>) failed to connect: ** (DBConnection.ConnectionError) ssl connect: TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA
 - {:tls_alert, {:unknown_ca, ~c"TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA\n"}}
16:37:24.465 [error] Postgrex.Protocol (#PID<0.167.0>) failed to connect: ** (DBConnection.ConnectionError) ssl connect: TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA
 - {:tls_alert, {:unknown_ca, ~c"TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA\n"}}
16:37:24.465 [error] Postgrex.Protocol (#PID<0.168.0>) failed to connect: ** (DBConnection.ConnectionError) ssl connect: TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA
 - {:tls_alert, {:unknown_ca, ~c"TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA\n"}}
16:37:24.586 [error] Postgrex.Protocol (#PID<0.204.0>) failed to connect: ** (DBConnection.ConnectionError) ssl connect: TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA
 - {:tls_alert, {:unknown_ca, ~c"TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA\n"}}
** (MatchError) no match of right hand side value: {:error, "killed"}
    (plausible 0.0.1) lib/plausible_release.ex:47: anonymous fn/2 in Plausible.Release.createdb/0
    (elixir 1.16.0) lib/enum.ex:2528: Enum."-reduce/3-lists^foldl/2-0-"/3
    (plausible 0.0.1) lib/plausible_release.ex:46: Plausible.Release.createdb/0
    nofile:1: (file)
    (stdlib 5.2) erl_eval.erl:750: :erl_eval.do_apply/7
16:37:24.589 [error] GenServer Plausible.IngestRepo terminating
** (MatchError) no match of right hand side value: {:error, "killed"}
    (plausible 0.0.1) lib/plausible_release.ex:47: anonymous fn/2 in Plausible.Release.createdb/0
    (elixir 1.16.0) lib/enum.ex:2528: Enum."-reduce/3-lists^foldl/2-0-"/3
    (plausible 0.0.1) lib/plausible_release.ex:46: Plausible.Release.createdb/0
    nofile:1: (file)
    (elixir 1.16.0) src/elixir.erl:405: :elixir.eval_external_handler/3
    (stdlib 5.2) erl_eval.erl:750: :erl_eval.do_apply/7
    (elixir 1.16.0) src/elixir.erl:378: :elixir.eval_forms/4
    (elixir 1.16.0) lib/module/parallel_checker.ex:112: Module.ParallelChecker.verify/1
Last message: {:EXIT, #PID<0.98.0>, {{:badmatch, {:error, "killed"}}, [{Plausible.Release, :"-createdb/0-fun-0-", 2, [file: ~c"lib/plausible_release.ex", line: 47]}, {Enum, :"-reduce/3-lists^foldl/2-0-", 3, [file: ~c"lib/enum.ex", line: 2528]}, {Plausible.Release, :createdb, 0, [file: ~c"lib/plausible_release.ex", line: 46]}, {:elixir_eval, :__FILE__, 1, [file: ~c"nofile", line: 1]}, {:elixir, :eval_external_handler, 3, [file: ~c"src/elixir.erl", line: 405]}, {:erl_eval, :do_apply, 7, [file: ~c"erl_eval.erl", line: 750]}, {:elixir, :eval_forms, 4, [file: ~c"src/elixir.erl", line: 378]}, {Module.ParallelChecker, :verify, 1, [file: ~c"lib/module/parallel_checker.ex", line: 112]}]}}
16:37:24.589 [error] GenServer Plausible.Repo terminating
** (MatchError) no match of right hand side value: {:error, "killed"}
    (plausible 0.0.1) lib/plausible_release.ex:47: anonymous fn/2 in Plausible.Release.createdb/0
    (elixir 1.16.0) lib/enum.ex:2528: Enum."-reduce/3-lists^foldl/2-0-"/3
    (plausible 0.0.1) lib/plausible_release.ex:46: Plausible.Release.createdb/0
    nofile:1: (file)
    (elixir 1.16.0) src/elixir.erl:405: :elixir.eval_external_handler/3
    (stdlib 5.2) erl_eval.erl:750: :erl_eval.do_apply/7
    (elixir 1.16.0) src/elixir.erl:378: :elixir.eval_forms/4
    (elixir 1.16.0) lib/module/parallel_checker.ex:112: Module.ParallelChecker.verify/1
Last message: {:EXIT, #PID<0.98.0>, {{:badmatch, {:error, "killed"}}, [{Plausible.Release, :"-createdb/0-fun-0-", 2, [file: ~c"lib/plausible_release.ex", line: 47]}, {Enum, :"-reduce/3-lists^foldl/2-0-", 3, [file: ~c"lib/enum.ex", line: 2528]}, {Plausible.Release, :createdb, 0, [file: ~c"lib/plausible_release.ex", line: 46]}, {:elixir_eval, :__FILE__, 1, [file: ~c"nofile", line: 1]}, {:elixir, :eval_external_handler, 3, [file: ~c"src/elixir.erl", line: 405]}, {:erl_eval, :do_apply, 7, [file: ~c"erl_eval.erl", line: 750]}, {:elixir, :eval_forms, 4, [file: ~c"src/elixir.erl", line: 378]}, {Module.ParallelChecker, :verify, 1, [file: ~c"lib/module/parallel_checker.ex", line: 112]}]}}
```

With SSL and `DATABASE_CACERTFILE` defined:

```
./createdb.sh 
Loading plausible..
Starting dependencies..
Starting repos..
create Plausible.Repo database if it doesn't exist
16:35:16.600 [error] Postgrex.Protocol (#PID<0.167.0>) failed to connect: ** (DBConnection.ConnectionError) ssl connect: TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA
 - {:tls_alert, {:unknown_ca, ~c"TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA\n"}}
16:35:16.600 [error] Postgrex.Protocol (#PID<0.185.0>) failed to connect: ** (DBConnection.ConnectionError) ssl connect: TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA
 - {:tls_alert, {:unknown_ca, ~c"TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA\n"}}
16:35:16.600 [error] Postgrex.Protocol (#PID<0.168.0>) failed to connect: ** (DBConnection.ConnectionError) ssl connect: TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA
 - {:tls_alert, {:unknown_ca, ~c"TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA\n"}}
16:35:16.682 [error] Postgrex.Protocol (#PID<0.204.0>) failed to connect: ** (DBConnection.ConnectionError) ssl connect: TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA
 - {:tls_alert, {:unknown_ca, ~c"TLS client: In state wait_cert at ssl_handshake.erl:2133 generated CLIENT ALERT: Fatal - Unknown CA\n"}}
** (MatchError) no match of right hand side value: {:error, "killed"}
    (plausible 0.0.1) lib/plausible_release.ex:47: anonymous fn/2 in Plausible.Release.createdb/0
    (elixir 1.16.0) lib/enum.ex:2528: Enum."-reduce/3-lists^foldl/2-0-"/3
    (plausible 0.0.1) lib/plausible_release.ex:46: Plausible.Release.createdb/0
    nofile:1: (file)
    (stdlib 5.2) erl_eval.erl:750: :erl_eval.do_apply/7
```

DATABASE_URL=postgresql://plausible:XXX@plausible-db.abc.us-east-1.rds.amazonaws.com:5432/plausible?ssl=true
DATABASE_CACERTFILE=/tmp/us-east-1-bundle.pem

**Comment by ruslandoga** [comment]
👋 @guatedude2 

Since the error is the same in both cases, I wonder if your could double check the env vars in the container?

```console
$ docker compose exec plausible env | grep DATABASE
```

**Comment by guatedude2** [comment]
> 👋 @guatedude2
> 
> Since the error is the same in both cases, I wonder if you could double check the env vars (and also cacertfile) in the container?
> 
> ```
> $ docker compose exec plausible env | grep DATABASE
> $ docker compose exec plausible stat /tmp/us-east-1-bundle.pem
> ```
> 
> Note that changes in `env_file` are not picked up automatically by Docker Compose and the container needs to be re-created for them to take place.

This is what comes back as:
```
DATABASE_URL=postgresql://plausible:xxxxxxxxxx@plausible.xxxxxxxxxxx.us-east-1.rds.amazonaws.com:5432/plausible?ssl=false
```
and 
```
  File: /tmp/us-east-1-bundle.pem
  Size: 4576      	Blocks: 16         IO Block: 4096   regular file
Device: 99h/153d	Inode: 3788720     Links: 1
Access: (0644/-rw-r--r--)  Uid: (  999/plausible)   Gid: (65533/ nogroup)
Access: 2024-09-19 19:20:38.074325013 +0000
Modify: 2024-09-20 19:54:53.962794002 +0000
Change: 2024-09-20 19:54:53.962794002 +0000
```

**Pull request closed without merging**